### PR TITLE
DPDFReduction: calibrate with white-beam vanadium

### LINKED ISSUE #19086: DPDFReduction: calibrate with white-beam vanadium
Calibration for detector efficiency should be done with white-beam vanadium.

The below script obtained a monocromatic vanadium from a white-beam vanadium in order to correct for detector efficiency. The same steps must be applied to our event data files:

```
LoadNexusMonitors(Filename='/SNS/ARCS/IPTS-18468/0/88358/NeXus/ARCS_88358_event.nxs', OutputWorkspace='__MonWS')
Ei, T0 = GetEiT0atSNS(MonitorWorkspace='__MonWS', IncidentEnergyGuess='35')
LoadEventNexus(Filename='/SNS/ARCS/IPTS-18468/0/88358/NeXus/ARCS_88358_event.nxs', OutputWorkspace='__IWS')
CorrectLogTimes(Workspace='__IWS')
LoadNexus(Filename='/SNS/ARCS/shared/autoreduce/vanadium_files/van88344.nxs', OutputWorkspace='__VAN')
DgsReduction(SampleInputWorkspace='__IWS',
                       SampleInputMonitorWorkspace='__MonWS',
                      IncidentEnergyGuess=Ei, 
                      UseIncidentEnergyGuess='1', 
                      TimeZeroGuess=T0,
                       EnergyTransferRange='-17.5,0.35,33.25',
                       IncidentBeamNormalisation='ByCurrent',
                        DetectorVanadiumInputWorkspace='__VAN',
                       UseProcessedDetVan='1',
                       OutputWorkspace='__OWS')
```

The division of our data by a monocromatic vanadium has the objective of obtaining absolute units for our scattering cross section, not to correct for detector efficency.

### PR DESCRIPTION
**Pull request by jmborr**
1. Checks vanadium is white-beam
2. Mask workspace before call to DGSReduction.
3. Passed properties to DGSReduction are those typically passed on to ARCS reduction.

**To test:**
Try to run the following script:
```
DPDFreduction(RunNumbers="88708-88711",
    Vanadium="/SNS/ARCS/IPTS-18468/shared/autoreduce/ARCS_88358_autoreduced.nxs",
    EmptyCanRunNumbers="88781",
    EnergyBins=1.5,
    MomentumTransferBins=0.1,
    OutputWorkspace="slices_QE")
```
You should be replied that the vanadium is not white-beam. Now run with a white-beam vanadium:

```
DPDFreduction(RunNumbers="88708-88711",
    Vanadium="/SNS/ARCS/shared/autoreduce/vanadium_files/van88344.nxs",
    EmptyCanRunNumbers="88781",
    EnergyBins=1.5,
    MomentumTransferBins=0.1,
    OutputWorkspace="slices_QE")
```

The reduction will produce workspace `slices_QE`.

Fixes #19086.

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/DPDFreduction.py [hunk 1]
```diff
@@ -370,6 +390,32 @@ def PyExec(self):
...
+        :return: None
+        """
+        # Find out the files for each run
+        load_algorithm = AlgorithmManager.createUnmanaged("Load")
+        load_algorithm.initialize()
+        load_algorithm.setPropertyValue('Filename', str(run_numbers))
+        files = (load_algorithm.getProperty('Filename').value)[0]
+        if not isinstance(files, list):
+            # run_numbers represents one file only
+            api.Load(Filename=files, LoadMonitors=True, OutputWorkspace=data_name)
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/DPDFreduction.py]: You can use `LoadEventNexus` from here on, if you only want this to work on ARCS. Leave `Load` if you want this to work at other facilities

File: Framework/PythonInterface/plugins/algorithms/DPDFreduction.py [hunk 2]
```diff
@@ -131,50 +131,51 @@ def PyExec(self):
...
-            # Load several event files into a single workspace. The nominal incident
-            # energy should be the same to avoid difference in energy resolution
-            api.Load(Filename=self._runs, OutputWorkspace=wn_data)
-
             # Load the vanadium file, assumed to be preprocessed, meaning that
             # for every detector all events within a particular wide wavelength
             # range have been rebinned into a single histogram
             api.Load(Filename=self._vanfile, OutputWorkspace=wn_van)
+            # Check for white-beam vanadium, true if the vertical chopper is absent (vChTrans==2)
+            if mtd[wn_van].run().getProperty('vChTrans').value[0] != 2:
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/DPDFreduction.py]: This works only on ARCS and SEQ. At some point in the code you MUST check if the instrument is ARCS. I would do it as part of the `_load` function. Just raise a `RuntimeError` or a `ValueError` or a `NotImplementedError`

File: Framework/PythonInterface/plugins/algorithms/DPDFreduction.py [hunk 3]
```diff
@@ -131,50 +131,50 @@ def PyExec(self):
...
-        wn_ec_data = prefix + 'ec_data'
-        wn_ec_reduced = prefix + 'ec_reduced'
-        wn_ec_ste = prefix + 'ec_S_theta_E'
+        wn_ec_data = prefix + 'ec_data'  # Accumulated empty can data
+        wn_ec_data_mon = prefix + 'ec_data_monitors'  # Accumulated monitors for empty can
+        wn_ec_reduced = prefix + 'ec_reduced'  # empty can data after DGSReduction
+        wn_ec_ste = prefix + 'ec_S_theta_E'  # empty can data after grouping by theta angle
 
         datasearch = config["datasearch.searcharchive"]
         if datasearch != "On":
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/DPDFreduction.py]: don't need an `if`. And move the other config options (instrument and facility) here, so they are grouped and just next to the code that uses those

**Review by AndreiSavici — CHANGES_REQUESTED** [review]

**Comment by jmborr** [comment]
@AndreiSavici I just reduced with empty array for input property `MomentumTransferBins` and it went through OK.

**Review by AndreiSavici — APPROVED** [review]

**Comment by AndreiSavici** [comment]
Looks fine. :shipit:

**Comment by jmborr** [comment]
@AndreiSavici @peterfpeterson Thanks!!

**Pull request merged**